lib/src/Pos/Block/Pure.hs

```diff
@@ -63,7 +62,10 @@ data VerifyHeaderParams ssc = VerifyHeaderParams
       -- ^ Maximal allowed header size. It's applied to 'BlockHeader'.
     , vhpVerifyNoUnknown :: !Bool
       -- ^ Check that header has no unknown attributes.
-    } deriving (Show, Eq)
+    }
+
+deriving instance Eq BlockHeader => Eq VerifyHeaderParams
```

Comment:
I really don't get why `Eq BlockHeader =>` and such stuff may be needed. `BlockHeader` is a concrete type and there is an `Eq` instance for it (we import `Pos.Block.Core`, `Pos.Core`, `Universum` here).

Reply:
I haven't investigated it because I wanted the code to compile and that's all. I'll fix it in part II.